src/app/PurchasePanel.tsx

```diff
@@ -0,0 +1,205 @@
+"use client";
+
+import {
+  PrePurchaseFailureReason,
+  PrePurchaseCheckResponse,
+  SaleEligibility,
+  EntitySetupState,
+  AllocationPermit,
+} from "@echoxyz/sonar-core";
+import { useEffect, useState } from "react";
+import { sonarConfig } from "./config";
+import { useSonarClient, useSonarEntity } from "@echoxyz/sonar-react";
+import { useAccount } from "wagmi";
+
+function PrePurchaseCheckState({
+  prePurchaseCheckState,
+}: {
+  prePurchaseCheckState: PrePurchaseCheckResponse;
+}) {
+  let stateDescription;
+  let stateFgColor;
+  let stateBgColor;
+  if (
+    prePurchaseCheckState.ReadyToPurchase ||
+    (!prePurchaseCheckState.ReadyToPurchase &&
+      prePurchaseCheckState.FailureReason ==
+        PrePurchaseFailureReason.REQUIRES_LIVENESS)
+  ) {
+    stateFgColor = "text-green-500";
+    stateBgColor = "bg-green-50";
+    stateDescription = "You are ready to purchase";
+  } else if (
+    prePurchaseCheckState.FailureReason ==
+    PrePurchaseFailureReason.MAX_WALLETS_USED
+  ) {
+    stateFgColor = "text-amber-500";
+    stateBgColor = "bg-amber-50";
+    stateDescription =
+      "Maximum number of wallets reached. This entity cannot use the connected wallet to commit funds to this sale. Please use a previous wallet or select a different investing entity.";
+  } else if (
+    prePurchaseCheckState.FailureReason ==
+    PrePurchaseFailureReason.NO_RESERVED_ALLOCATION
+  ) {
+    stateFgColor = "text-amber-500";
+    stateBgColor = "bg-amber-50";
+    stateDescription =
+      "No reserved allocation. The connected wallet has no reserved allocation for this sale. Please connect a different wallet.";
+  } else {
+    stateFgColor = "text-red-500";
+    stateBgColor = "bg-red-50";
+    stateDescription = "Unknown failure reason. Please contact support.";
+  }
```

Comment:
this is very hard to read. would move this to a helper function and have returns instead of overrides + list out all cases seperately. e.g. something like
```
type stateConfig {
  fgColor: string;
  bgColor: string;
  description: string;
}

okConfig(msg) => {
  fgCol: green,
  bgCol: green,
  description: msg
}

stateConfig(resp): config => {
  if (readyToPurchase) {
    return okConfig(".....")
  }

  switch (prePurchaseCheckState.FailureReason:
  case PrePurchaseFailureReason.REQUIRES_LIVENESS: 
       return  okConfig("go do liveness")
  case  PrePurchaseFailureReason.MAX_WALLETS_USED:
       return warningConfig("....")
  ....


  return errorConfig("...")
}